packages/office-ui-fabric-react/src/utilities/DraggableZone/DraggableZone.tsx

```diff
@@ -0,0 +1,335 @@
+import * as React from 'react';
+import { IDraggableZoneProps, ICoordinates, IDragData } from './DraggableZone.types';
+import { getClassNames } from './DraggableZone.styles';
+import { on } from '../../Utilities';
+
+export interface IDraggableZoneState {
+  isDragging: boolean;
+  position: ICoordinates;
+  lastPosition?: ICoordinates;
+}
+
+const eventMapping = {
+  touch: {
+    start: 'touchstart',
+    move: 'touchmove',
+    stop: 'touchend'
+  },
+  mouse: {
+    start: 'mousedown',
+    move: 'mousemove',
+    stop: 'mouseup'
+  }
+};
+
+// These are needed so that we can generalize the events
+// and so we have access to clientX and clientY in the touch events
+type MouseTouchEvent<T> = React.MouseEvent<T> & React.TouchEvent<T> & Event;
+
+export class DraggableZone extends React.PureComponent<IDraggableZoneProps, IDraggableZoneState> {
+  private _touchId?: number;
```

Comment:
I was a little wrong here @jspurlin ; I think you need to use `React.Component`, and not `PureComponent`. Whenever you have children in a component, those children are constantly mutated:

```tsx
<DraggableZone>{content}</DraggableZone>
```

So let's say a parent has an interval setup to re-render every second. Even if DraggableZone's children are identical, `children` will be a non-immutable object, causing PureComponent's shallow diffing to always return true for `shouldComponentUpdate`.

The requirements for using `PureComponent` are:

1. No children
2. All props are (or could be) immutable values

Reply:
Yeah, good point, updated